adam-core/src/main/scala/org/bdgenomics/adam/serialization/ADAMKryoRegistrator.scala

```diff
@@ -283,9 +283,11 @@ class ADAMKryoRegistrator extends KryoRegistrator with Logging {
     try {
       kryo.register(Class.forName("org.apache.spark.internal.io.FileCommitProtocol$TaskCommitMessage"))
       kryo.register(Class.forName("org.apache.spark.sql.execution.datasources.FileFormatWriter$WriteTaskResult"))
+      kryo.register(Class.forName("org.apache.spark.sql.execution.datasources.BasicWriteTaskStats"))
+      kryo.register(Class.forName("org.apache.spark.sql.execution.datasources.ExecutedWriteSummary"))
     } catch {
       case cnfe: java.lang.ClassNotFoundException => {
-        log.info("Did not find Spark internal class. This is expected for Spark 1.")
+        log.info("Did not find Spark internal class. This is expected for earlier Spark versions.")
```

Comment:
Can you say spark < 2.3.0 or is that incorrect?

Reply:
So the first two `kryo.register` in this `try`  were already there from a Spark1 to Spark2 incompatibility and the two I add i this PR are for a <2.3 to a 2.3 incompatibility.
But I think just saying <2.3 as you suggest is probably fine, will do unless @heuermh comments otherwise.

Reply:
TBH the errors with Spark 1 won't happen any more, since we don't support it.  There are a few other places in this file that could be updated as well, but that is a low priority.